tests/integration/test_load_and_serve_agent_a2a.py

```diff
@@ -0,0 +1,151 @@
+import logging
+import os
+import subprocess
+from datetime import datetime
+from pathlib import Path
+from typing import Any
+from datetime import datetime
+
+import pytest
+from litellm.utils import validate_environment
+
+import any_agent.serving
+from any_agent import AgentConfig, AgentFramework, AnyAgent, TracingConfig
+from any_agent.config import MCPStdio, ServingConfig
+
+from opentelemetry.sdk.trace.export.in_memory_span_exporter import (
+    InMemorySpanExporter,
+)
+from opentelemetry.sdk.trace import ReadableSpan
+
+from common.types import (
+    Message,
+    TaskState,
+)
+
+if any_agent.serving.serving_available:
+    from common.client import A2AClient
+    from common.types import Message, TextPart
+
+FORMAT = "%(message)s"
+logging.basicConfig(
+    format=FORMAT,
+    datefmt="[%X]",
+)
+logger = logging.getLogger("any_agent_test")
+logger.setLevel(logging.DEBUG)
+
+def check_uvx_installed() -> bool:
+    """The integration tests requires uvx"""
+    try:
+        result = subprocess.run(  # noqa: S603
+            ["uvx", "--version"],  # noqa: S607
+            capture_output=True,
+            check=True,
+        )
+        return True if result.returncode == 0 else False  # noqa: TRY300
+    except Exception:
+        return False
```

Comment:
Does this code exist elsewhere and can we use that instead of copying here?

Reply:
Yes, I'll make a lib out of this. Basically this is copied from the multiagent test, so now we can extract the common parts to make a util lib. But I think it's a refactoring to be done in a follow-up PR.